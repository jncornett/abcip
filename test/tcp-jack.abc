a ( syn )
b ( syn, ack )
a ( ack )

a ( ack, pay="VWXYZ", jump=21 )
a ( ack, pay="ABCDEFG", jump=-26 )
b ( ack )

a ( ack, pay="OPQRSTU", jump=7 )
a ( ack, pay="HIJKLMN", jump=-14 )
b ( ack, jack=12 )

a ( fin, ack, jump=12 )
b ( fin, ack )
a ( ack )

