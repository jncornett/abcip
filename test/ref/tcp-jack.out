IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [S], cksum 0xb3d9 (correct), seq 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 5cbc 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0008 0000 0001 0000 0000 5002  ..............P.
	0x0030:  2000 b3d9 0000                           ......
IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [S.], cksum 0xb3c7 (correct), seq 1, ack 2, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 5cbc 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0008 bdec 0000 0001 0000 0002 5012  ..............P.
	0x0030:  2000 b3c7 0000                           ......
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0xb3c8 (correct), ack 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0002 0000 4006 5cbb 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0008 0000 0002 0000 0002 5010  ..............P.
	0x0030:  2000 b3c8 0000                           ......
IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 45)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0xaafd (correct), seq 22:27, ack 1, win 8192, length 5
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  002d 0003 0000 4006 5cb5 0a01 0203 0a09  .-....@.\.......
	0x0020:  0807 bdec 0008 0000 0017 0000 0002 5010  ..............P.
	0x0030:  2000 aafd 0000 5657 5859 5a              ......VWXYZ
IP (tos 0x0, ttl 64, id 4, offset 0, flags [none], proto TCP (6), length 47)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0xa2f4 (correct), seq 1:8, ack 1, win 8192, length 7
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  002f 0004 0000 4006 5cb2 0a01 0203 0a09  ./....@.\.......
	0x0020:  0807 bdec 0008 0000 0002 0000 0002 5010  ..............P.
	0x0030:  2000 a2f4 0000 4142 4344 4546 47         ......ABCDEFG
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [.], cksum 0xb3c1 (correct), ack 8, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0002 0000 4006 5cbb 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0008 bdec 0000 0002 0000 0009 5010  ..............P.
	0x0030:  2000 b3c1 0000                           ......
IP (tos 0x0, ttl 64, id 5, offset 0, flags [none], proto TCP (6), length 47)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0x6abc (correct), seq 15:22, ack 1, win 8192, length 7
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  002f 0005 0000 4006 5cb1 0a01 0203 0a09  ./....@.\.......
	0x0020:  0807 bdec 0008 0000 0010 0000 0002 5010  ..............P.
	0x0030:  2000 6abc 0000 4f50 5152 5354 55         ..j...OPQRSTU
IP (tos 0x0, ttl 64, id 6, offset 0, flags [none], proto TCP (6), length 47)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0x86d8 (correct), seq 8:15, ack 1, win 8192, length 7
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  002f 0006 0000 4006 5cb0 0a01 0203 0a09  ./....@.\.......
	0x0020:  0807 bdec 0008 0000 0009 0000 0002 5010  ..............P.
	0x0030:  2000 86d8 0000 4849 4a4b 4c4d 4e         ......HIJKLMN
IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [.], cksum 0xb3ae (correct), ack 27, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0003 0000 4006 5cba 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0008 bdec 0000 0002 0000 001c 5010  ..............P.
	0x0030:  2000 b3ae 0000                           ......
IP (tos 0x0, ttl 64, id 7, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [F.], cksum 0xb3ad (correct), seq 27, ack 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0007 0000 4006 5cb6 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0008 0000 001c 0000 0002 5011  ..............P.
	0x0030:  2000 b3ad 0000                           ......
IP (tos 0x0, ttl 64, id 4, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [F.], cksum 0xb3ac (correct), seq 1, ack 28, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0004 0000 4006 5cb9 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0008 bdec 0000 0002 0000 001d 5011  ..............P.
	0x0030:  2000 b3ac 0000                           ......
IP (tos 0x0, ttl 64, id 8, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0xb3ac (correct), ack 2, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0008 0000 4006 5cb5 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0008 0000 001d 0000 0003 5010  ..............P.
	0x0030:  2000 b3ac 0000                           ......
